runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java

```diff
@@ -188,6 +192,17 @@ private void priceLookup(boolean exactMatch)
 			return;
 		}
 
+		// If the input contains characters not part of the pattern
+		// then show an error. This is to prevent bad lookups that can
+		// cause an npe and therefore soft-crash the plugin
+		if (!PATTERN.matcher(lookup).matches())
+		{
+			searchBox.setIcon(ERROR_ICON);
+			errorPanel.setContent("Error fetching results", "The input contains invalid characters. Please remove them.");
```

Comment:
What characters are invalid? How should users know what is allowed?

Perhaps `"Invalid input detected! Please ensure your search contains only letters, numbers, spaces, underscores, apostrophes, and parentheses."` would help clarify the requirements?